### PR TITLE
Add Datadog dashboard SLO widget support

### PR DESCRIPTION
**Pull request by mbarrien**
Adds support for the SLO widget in a Datadog dashboard. Can't find an official source of text on Datadog's API website, so created my own documentation not from official sources.

### CODE REVIEW

**Comment by mbarrien** [comment]
Ping @nmuesch ?

**Comment by KeisukeYamashita** [comment]
:+1:

I want SLO widget too in this terraform provider

**Comment by mbarrien** [comment]
Rebased against master, which now already has a version of go-datadog-api with the necessary APIs, so no more updates to go.mod in this PR. Please review and merge!

File: datadog/resource_datadog_dashboard.go [hunk 1]
```diff
@@ -2940,6 +2957,120 @@ func buildTerraformScatterplotRequest(datadogScatterplotRequest *datadog.Scatter
...
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"title_align": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"view_type": {
+			Type:     schema.TypeString,
+			Optional: true,
```
Comment (platinummonkey) [inline datadog/resource_datadog_dashboard.go]: `Required: true,`

File: datadog/resource_datadog_dashboard.go [hunk 2]
```diff
@@ -2940,6 +2957,120 @@ func buildTerraformScatterplotRequest(datadogScatterplotRequest *datadog.Scatter
...
+		},
+		"slo_id": {
+			Type:     schema.TypeString,
+			Optional: true,
```
Comment (platinummonkey) [inline datadog/resource_datadog_dashboard.go]: `Required: true,`

File: datadog/resource_datadog_dashboard.go [hunk 3]
```diff
@@ -2940,6 +2957,120 @@ func buildTerraformScatterplotRequest(datadogScatterplotRequest *datadog.Scatter
...
+		},
+		"show_error_budget": {
+			Type:     schema.TypeBool,
+			Optional: true,
+		},
+		"view_mode": {
+			Type:     schema.TypeString,
+			Optional: true,
```
Comment (platinummonkey) [inline datadog/resource_datadog_dashboard.go]: `Required: true,`

File: datadog/resource_datadog_dashboard.go [hunk 4]
```diff
@@ -2940,6 +2957,120 @@ func buildTerraformScatterplotRequest(datadogScatterplotRequest *datadog.Scatter
...
+		},
+		"time_windows": {
+			Type:     schema.TypeList,
+			Optional: true,
```
Comment (platinummonkey) [inline datadog/resource_datadog_dashboard.go]: `Required: true,`

File: datadog/resource_datadog_dashboard_test.go [hunk 5]
```diff
@@ -702,6 +712,15 @@ func TestAccDatadogDashboard_update(t *testing.T) {
 					resource.TestCheckResourceAttr("datadog_dashboard.ordered_dashboard", "widget.12.group_definition.0.widget.1.alert_graph_definition.0.viz_type", "toplist"),
 					resource.TestCheckResourceAttr("datadog_dashboard.ordered_dashboard", "widget.12.group_definition.0.widget.1.alert_graph_definition.0.title", "Alert Graph"),
 					resource.TestCheckResourceAttr("datadog_dashboard.ordered_dashboard", "widget.12.group_definition.0.widget.1.alert_graph_definition.0.time.live_span", "1h"),
+					// Service Level Objective widget
+					resource.TestCheckResourceAttr("datadog_dashboard.ordered_dashboard", "widget.13.scatterplot_definition.0.title", "Widget Title"),
```
Comment (bkabrda) [inline datadog/resource_datadog_dashboard_test.go]: After adding this, you'll also need to increment the number of widgets on line 530 (from 13 to 14).

Comment (bkabrda) [inline datadog/resource_datadog_dashboard_test.go]: In all these steps, you want to look for `service_level_objective_definition`, not `scatterplot_definition`.

File: website/docs/r/dashboard.html.markdown [hunk 6]
```diff
@@ -719,6 +730,15 @@ Nested `widget` blocks have the following structure:
         - `title_size`: (Optional) The size of the widget's title. Default is 16.
         - `title_align`: (Optional) The alignment of the widget's title. One of "left", "center", or "right".
         - `time`: (Optional) Nested block describing the timeframe to use when displaying the widget. The structure of this block is described [below](dashboard.html#nested-widget-time-blocks).
+  - `service_level_objective_definition`: The definition for a Service Level Objective widget. Exactly one nested block is allowed with the following structure:
```
Comment (bkabrda) [inline website/docs/r/dashboard.html.markdown]: After updating the fields pointed out by @platinummonkey to be Required, please also update the `(Optional)` => `(Required)` for these.

File: datadog/resource_datadog_dashboard.go (hunk 1 above)
Comment (mbarrien) [reply datadog/resource_datadog_dashboard.go]: I have made anything that was marked as "omitempty" in https://github.com/zorkian/go-datadog-api/pull/277/files as Optional. Should I ignore that API and allow these to mismatch in what is required and what is optional? (This applies throught this PR).

Comment (platinummonkey) [reply datadog/resource_datadog_dashboard.go]: for the SLO widget yes, these are required. 🙏

**Comment by KeisukeYamashita** [comment]
@bkabrda We fix according to your comments. Please review and merge:)

**Comment by bkabrda** [comment]
Code looks great, all comments have been addressed and tests are passing. LGTM, merging. Thank you very much for the contribution!

**Pull request merged**